frontend/src/lib/utils.ts

```diff
@@ -180,3 +184,24 @@ export function isValidFormId(formId?: string): formId is string {
 export function isInForm(widget: { formId?: string }): boolean {
   return isValidFormId(widget.formId)
 }
+
+export enum LabelVisibilityOptions {
+  Visible,
+  Hidden,
+  Collapsed,
+}
+
+export function labelVisibilityProtoValueToEnum(
+  value: LabelVisibilityMessageProto.LabelVisibilityOptions | null | undefined
+): LabelVisibilityOptions {
+  switch (value) {
+    case LabelVisibilityMessageProto.LabelVisibilityOptions.VISIBLE:
+      return LabelVisibilityOptions.Visible
+    case LabelVisibilityMessageProto.LabelVisibilityOptions.HIDDEN:
+      return LabelVisibilityOptions.Hidden
+    case LabelVisibilityMessageProto.LabelVisibilityOptions.COLLAPSED:
+      return LabelVisibilityOptions.Collapsed
+    default:
+      return LabelVisibilityOptions.Visible
+  }
```

Comment:
This may be my rec for simplification, but it seems like we don't push protos outside of the main component itself, so it seemed like separating concerns of Proto Enum vs the JS Enum